client/tiny.js

```diff
@@ -160,18 +167,23 @@ function attachPublisher(session) {
     var remoteReady = offerReady
         .then(handle.sendJsep.bind(handle))
         .then(answer => conn.setRemoteDescription(answer.jsep));
+    showStatus(`Connecting WebRTC...`);
     var connectionReady = Promise.all([iceReady, localReady, remoteReady]);
     return connectionReady
-      .then(() => handle.sendMessage({ kind: "join", room_id: roomId, user_id: USER_ID, subscribe: { "notifications": true, "data": true }}))
+      .then(() => {
+        showStatus(`Joining room ${roomId}...`);
+        return handle.sendMessage({ kind: "join", room_id: roomId, user_id: USER_ID, subscribe: { "notifications": true, "data": true }});
+      })
       .then(reply => {
+        showStatus(`Joined room ${roomId}`);
         var occupants = reply.plugindata.data.response.users[roomId];
         for (var i = 0; i < occupants.length; i++) {
           if (occupants[i] !== USER_ID) {
             addUser(session, occupants[i]);
           }
         }
       })
-      .then(() => { return { handle, conn, channel, unreliableChannel }; });
+      .then(() => { handle, conn, channel, unreliableChannel });
```

Comment:
Not sure if you have a strong preference for the previous syntax.

Reply:
Don't care.